standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2802,6 +2806,7 @@ PartitionsResponse get_partitions_req(1:PartitionsRequest req)
 
   void add_replication_metrics(1: ReplicationMetricList replicationMetricList) throws(1:MetaException o1)
   ReplicationMetricList get_replication_metrics(1: GetReplicationMetricsRequest rqst) throws(1:MetaException o1)
+  GetOpenTxnsResponse get_open_txns_req(1: GetOpenTxnsRequest getOpenTxnsRequest)
```

Comment:
Why did you introduce a new HMS API method for this?

I would add a new type attribute to TxnInfo and use get_open_txns_info instead

Reply:
get_open_txns_info doesn't take any input params. We needed to exclude certain type of txns and only return the other open txns.
The other way could have been add a new field in TxnInfo with the TxnType and still return all the open txns, but filter on the client side.
Currently get_open_txns_info filters out the read txns without informing the client. So we thought it might be better to expose an explicit call to exclude specific txn types.

Reply:
Fair point